src/gr/uom/java/xmi/diff/MergeAttributeRefactoring.java

```diff
@@ -141,15 +136,15 @@ public boolean equals(Object obj) {
 	@Override
 	public List<CodeRange> rightSide() {
 		List<CodeRange> ranges = new ArrayList<CodeRange>();
-		ranges.add(newAttribute.codeRange()
+		ranges.add(newAttribute.getVariableDeclaration().codeRange()
 				.setDescription("new attribute declaration")
-				.setCodeElement(newAttribute.toString()));
+				.setCodeElement(newAttribute.getVariableDeclaration().toString()));
 		return ranges;
 	}
 
 	private Set<VariableDeclaration> getMergedVariables() {
 		if (mergedAttributes == null)
-			return null;
+			return Collections.emptySet();
 		return mergedAttributes.stream().map(UMLAttribute::getVariableDeclaration).filter(Objects::nonNull).collect(Collectors.toSet());
```

Comment:
Does this set preserve the original order?

Reply:
No, it doesn't. I need to change it. Can we assume that the result of getVariableDeclaration of UMLAttribute is not null?

Reply:
Yes, this is guaranteed. getVariableDeclaration is not null.

Reply:
Then I can remove the nonNull filter.